src/duckling/controls/json.component.ts

```diff
@@ -1,22 +1,68 @@
 import {
     Component,
-    Input
+    Input,
+    Output,
+    EventEmitter,
+    OnChanges,
+    SimpleChange,
+    AfterViewInit
 } from '@angular/core';
 
+import {immutableAssign} from '../util/model';
+
+import {ValidatedInput} from './index';
+
 /**
  * A Component used to display the json representing an arbitrary value.
  */
 @Component({
     selector: "dk-json",
-    template: "<div>{{valueAsJSON}}<div>"
+    directives: [ValidatedInput],
+    template: `
+        <dk-validated-input
+            label="JSON"
+            [value]="valueAsJSON"
+            [validator]="isValidJSON"
+            (validInput)="onValueChanged($event)">
+        </dk-validated-input>
+    `
 })
-export class JsonComponent {
+export class JsonComponent implements OnChanges {
     /**
      * The object that will be displayed as json.
      */
     @Input() value : any;
 
+    /**
+     * Emits when the json was changed with the new json
+     */
+    @Output() valueChanged = new EventEmitter<any>();
+
+    private _stringValue = "";
+
+    ngOnChanges(changes : {value? : SimpleChange}) {
+        if (changes.value.currentValue !== this.value) {
```

Comment:
This should be a deep equality check between currentValue and previous value. This check is the reason the component isn't updating.